pyproject.toml

```diff
@@ -49,37 +39,100 @@ classifiers = [
        "Programming Language :: Python :: 3 :: Only",
 ]
 
+[dependency-groups]
+dev = [
+    "pytest>=3.9",
+    "hypothesis>=6.28.0",
+    "pytest-trio>=0.8.0",
+    "pyright>=1.1.404",
+    "ruff>=0.12.10",
+    "tox>=4.28.4",
+    "tox-uv>=1.28.0",
+    "tzdata>=2025.2",
+    "pytest-xdist>=3.8.0",
+    "pytest-cov>=6.2.1",
+    "pre-commit>=4.3.0",
+]
+docs = [
+       "sphinx>=8.1.3; python_version >= '3.11'",
```

Comment:
add the python marker, since uv didn't manage to resolve for all supported python versions which we don't care about for building docs anyway.